### PR TITLE
Two-Function nodes

### PR DESCRIPTION
**Pull request by bencloward**
### Purpose of this PR
Two-Function nodes:

- Parallax Mapping
- Parallax Occlusion Mapping
- Compute Deformation
- Linear Blend Skinning
- Simple Noise
- Gradient Noise
- Voronoi

Also fixed a couple of issues with the SampleTexture2D node.

Please note that the follow nodes are not done, but Esme asked me to check them in anyway so that she can use them as a starting point for her work:
- Parallax Occlusion Mapping
- Compute Deformation
- Linear Blend Skinning
For each of these, I have created a separate Jira for Esme.

### CODE REVIEW

**Comment by bencloward** [comment]
Please note that the follow nodes are not done, but Esme asked me to check them in anyway so that she can use them as a starting point for her work:

Parallax Occlusion Mapping
Compute Deformation
Linear Blend Skinning
For each of these, I have created a separate Jira for Esme.

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/VoronoiNode.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    {
+        static string Name = "Voronoi";
+        static int Version = 1;
+        public static NodeDescriptor NodeDescriptor => new(
+            Version,
+            Name,
+            "VoronoiDeterministic",
+            functions: new FunctionDescriptor[] {
+                new(
+                    "Unity_Voronoi_RandomVector_Deterministic",
```
Comment (tracychen1234) [inline com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/VoronoiNode.cs:16]: `Shader error in 'Test': undeclared identifier 'Unity_Voronoi_RandomVector_Deterministic_float' at line 476 (on d3d11)` same as above

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/UV/ParallaxOcclusionMappingNode.cs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+            tooltip: "Creates a parallax effect that displaces a material's UVs and depth to create the illusion of depth.",
+            categories: new string[1] { "UV" },
+            synonyms: new string[1] { "pom" },
+            hasPreview: false,
+            selectableFunctions: new()
+            {
+                { "ParallaxOcclusionMappingRed", "Red" },
+                { "ParallaxOcclusionMappingGreen", "Green" },
+                { "ParallaxOcclusionMappingBlue", "Blue" },
+                { "ParallaxOcclusionMappingAlpha", "Alpha" }
```
Comment (tracychen1234) [inline com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/UV/ParallaxOcclusionMappingNode.cs:298]: a note of setting drop-down label in the future?

**Review by tracychen1234 — APPROVED** [review]
Looks good!

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/GradientNoiseNode.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    {
+        static string Name = "GradientNoise";
+        static int Version = 1;
+        public static NodeDescriptor NodeDescriptor => new(
+            Version,
+            Name,
+            "GradientNoiseDeterministic",
+            functions: new FunctionDescriptor[] {
+                new(
+                    "Unity_GradientNoise_Deterministic_Dir",
```
[the remark this answers is not in the document]
Comment (bencloward) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/GradientNoiseNode.cs:16]: This should now be fixed.

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/VoronoiNode.cs (hunk 1 above)
Comment (bencloward) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/VoronoiNode.cs:16]: This should now be fixed.

**Review by esmelusina — APPROVED** [review]
Looks good- we really need UVs to test these proper.

**Comment by bencloward** [comment]
Yes.  Yes, we do.

**Pull request merged**